bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVM.java

```diff
@@ -1840,9 +1840,16 @@ private static void execStoreOpcodes(Strand ctx, StackFrame sf, int opcode, int[
                 BNewArray list = Optional.of((BNewArray) sf.refRegs[i]).get();
                 long index = sf.longRegs[j];
                 BRefType refReg = sf.refRegs[k];
-                BType elementType = (list.getType().getTag() == TypeTags.ARRAY_TAG)
-                        ? ((BArrayType) list.getType()).getElementType()
-                        : ((BTupleType) list.getType()).getTupleTypes().get((int) index);
+                BType elementType;
+                if (list.getType().getTag() == TypeTags.ARRAY_TAG) {
+                    elementType = ((BArrayType) list.getType()).getElementType();
+                } else {
+                    BTupleType tuple = getTupleFieldType(ctx, (BTupleType) list.getType(), index);
```

Comment:
Isn't the method name `getTupleFieldType` a bit misleading? 

Also isn't the method more like a range check? How about updating it to return a `boolean` instead? 

Should we also consider refactoring `execListAddOperation` for tuples? Or we will be repeating the range check right?